ooni/templates/httpt.py

```diff
@@ -122,11 +122,28 @@ def addToReport(self, request, response=None, response_body=None, failure_string
 
             failure (instance): An instance of :class:twisted.internet.failure.Failure
         """
+        def _representHeaders(headers):
+            represented_headers = {}
+            for name, value in headers.getAllRawHeaders():
+                represented_headers[name] = value[0]
+            return represented_headers
+
+        def _representBody(body):
+            # XXX perhaps add support for decoding gzip in the future.
+            try:
+                body.replace('\0', '')
```

Comment:
This statement has no effect because replace() does not change the string inplace but rather returns a modified string. In any case, I am not sure it's a good idea to alter the body. Why would you like to do that?